From cd41302e0278c0d0f447855b24cb8b8969d2ab60 Mon Sep 17 00:00:00 2001
From: Robert Lehmann <robert.lehmann@sitec-systems.de>
Date: Thu, 9 Mar 2017 16:13:07 +0100
Subject: [PATCH 19/19] configs: score: Add scripts for mfg boot

Add a script for flashing the SPI NOR with a valid u-boot image. The u-boot image
is loaded from RAM. The script depends on the mfg tool which needs to load the
u-boot image into the defined RAM location.
Also add script for writing the fuses.

Remove some unused defines.

Ticket #1763
---
 include/configs/score.h | 21 ++++++++++++++-------
 1 file changed, 14 insertions(+), 7 deletions(-)

diff --git a/include/configs/score.h b/include/configs/score.h
index 0abaf5b..2b5e7bb 100644
--- a/include/configs/score.h
+++ b/include/configs/score.h
@@ -12,9 +12,12 @@
 
 #include <asm/arch/imx-regs.h>
 #include <linux/sizes.h>
-#include "mx6_common.h"
 #include <asm/imx-common/gpio.h>
 
+#define CONFIG_SYS_BOOTM_LEN SZ_32M
+
+#include "mx6_common.h"
+
 #define CONFIG_MX6
 #define CONFIG_ROM_UNIFIED_SECTIONS
 #define CONFIG_SYS_GENERIC_BOARD
@@ -69,7 +72,6 @@
 #undef CONFIG_BOOTM_PLAN9
 #undef CONFIG_BOOTM_RTEMS
 
-#define CONFIG_SCORE_NETWORK
 #ifdef CONFIG_SCORE_NETWORK
 /* Network configs */
 #define CONFIG_CMD_PING
@@ -111,9 +113,11 @@
 #define CONFIG_PMIC_I2C_SLAVE		0x8
 #endif
 
+#define CONFIG_CMD_ASKENV
+
 #define CONFIG_PCA953X
 #define CONFIG_CMD_PCA953X
-#define CONFIG_PCA953X_BUS_NUM 2
+#define CONFIG_PCA953X_BUS_NUM 1
 #define CONFIG_PCA953X_OUT_MASK 0xecdf
 #define CONFIG_PCA953X_OUT_INIT 0x8001
 #define CONFIG_CMD_PCA953X_INFO
@@ -132,7 +136,6 @@
 #define CONFIG_LOADADDR			0x80800000
 #define CONFIG_SYS_TEXT_BASE		0x87800000
 
-#define CONFIG_SYS_BOOTM_LEN SZ_32M
 
 #define CONFIG_SYS_AUXCORE_BOOTDATA 0x78000000 /* Set to QSPI2 B flash at default */
 #ifndef CONFIG_SYS_AUXCORE_FASTUP
@@ -169,9 +172,11 @@
         "root=/dev/ram0 " \
         "mtdparts=${mtdparts} "\
         "\0" \
-    "initrd_addr=0x83800000\0" \
-    "initrd_high=0xffffffff\0" \
-    "bootcmd_mfg=run mfgtool_args;bootm ${loadaddr} - ${fdt_addr};\0" \
+    "uboot_addr=0x83800000\0" \
+	"uboot_flash=crc32 83800000 100; askenv crcvalid \"Is CRC32 valid [y/n]\" 1; if test $crcvalid = \"y\"; then sf probe && sf erase 0x0 0x60000 && sf write ${uboot_addr} 0x400 0x60000; fi\0" \
+	"burning=fuse prog -y 0 5 0x2a000030 && fuse prog -y 0 6 0x10\0" \
+	"bootusb=run mfgtool_args;bootm ${loadaddr} - ${fdt_addr};\0" \
+	"bootcmd_mfg=false\0" \
 
 #define CONFIG_BOOT_COMMON \
     "fdt_addr=0x83000000\0" \
@@ -303,6 +308,7 @@
 #define CONFIG_CMD_CACHE
 #endif
 
+#if 0
 #define CONFIG_CMD_USB
 #define CONFIG_USB_EHCI
 #define CONFIG_USB_EHCI_MX6
@@ -313,5 +319,6 @@
 #define CONFIG_MXC_USB_PORTSC  (PORT_PTS_UTMI | PORT_PTS_PTW)
 #define CONFIG_MXC_USB_FLAGS   0
 #define CONFIG_USB_MAX_CONTROLLER_COUNT 2
+#endif
 
 #endif				/* __CONFIG_H */
-- 
2.1.4

